Tools/TestWebKitAPI/Tests/IPC/ConnectionTests.cpp

```diff
@@ -77,10 +85,19 @@ class MockConnectionClient final : public IPC::Connection::Client {
         return m_didClose;
     }
 
+    // Handler returns false if the message should be just recorded.
+    void setAsyncMessageHandler(Function<bool(MessageInfo&)>&& handler)
+    {
+        m_asyncMessageHandler = WTFMove(handler);
+    }
+
     // IPC::Connection::Client overrides.
     void didReceiveMessage(IPC::Connection&, IPC::Decoder& decoder) override
     {
-        m_messages.insert(0, { decoder.messageName(), decoder.destinationID() });
+        MessageInfo messageInfo { decoder.messageName(), decoder.destinationID() };
+        if (m_asyncMessageHandler && m_asyncMessageHandler(messageInfo))
+            return;
+        m_messages.insert(0, WTFMove(messageInfo));
```

Comment:
Again, this is testing code so probably doesn't matter but it seems like a good use case for a WTF::Deque, not a WTF::Vector when you have to prepend.